spec/steps/provision.fmf

```diff
@@ -97,3 +96,47 @@ example: |
             how: connect
             guest: hostname or ip address
             key: private-key-path
+
+/environment:
+    summary: Environment specification
+
+    description: |
+        As part of the ``provision`` step it is possible to
+        specify additional requirements for the testing
+        environment such as required amount of ``memory`` and
+        ``disk`` space.
+
+        FIXME: Much more details go here. For now let's start with
+        a couple of tangible examples.
+
+    example: |
+        # Simple use cases simple
+        memory: 8 GB
+        disk: 500 GB
+
+        # Processor-related stuff grouped together
+        cpu:
+            processors: 2
+            cores: 16
+            model: 37
+
+        # Operators at the start of the value
+        memory: '> 8 GB'
+        memory: '>= 8 GB'
+        memory: '< 8 GB'
+        memory: '<= 8 GB'
+
+        # By default exact value expected, these are equivalent:
+        model: 37
+        model: = 37
+
+        # Enable regular expression search
+        model-name: =~ .*AMD.*
```

Comment:
we should mention that not full extent of regular expressions might be supported everywhere, but at least `.*` should be supported on all infrastructures.

Reply:
Added a new paragraph mentioning this.